### ISSUE TITLE
forbid sending rich text messages with more than 19 mb atts

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#2538]
- it's hard on the recipient, especially on mobile
- gmail may not accept it due to size problems (over 20mb or so) due to several layers of encoding (size bloat)

**Comment by tomholub** [comment]
Interestingly didn't run into this again yet, but is likely still a problem. May need to have a more hollistic look into encoding, attachments, rich text.

**Comment by tomholub** [comment]
would become irrelevant if #3032 is resolved by Gmail

**Comment by tomholub** [comment]
@martgil please try, on firefox, when switched to "Rich text" mode, what is the largest size of attachment that gets accepted and also successfully sent. Does it work with a 25mb attachment? Or is it less?

**Comment by martgil** [comment]
These are what I noticed when sending an email with attachment in "Rich text" mode.

![image](https://user-images.githubusercontent.com/46025304/229971074-f336768f-73e1-4f02-b350-b72788fd1724.png)
For Figure 1: When attaching a 21mb image file in rich text mode, the FlowCrypt secure compose window have no problem when including the image. However, an error will occur when sending the secure email with attachment saying "couldn't send the email because the attachment is too large. For comparison, I tried to send the same 21mb image file in non-rich text mode and the email went through.
![image](https://user-images.githubusercontent.com/46025304/229971680-bdf63442-bb1e-45e1-87ac-04af339eb235.png)
(figure no. 1)

For Figure 2: In rich text mode, when attaching 2 or more images with 21mb file size each, a warning prompt will appear. While it is a correct, if we watched video no. 1, the attachment prompt continues to show a warning prompt but the images get appended in the Rich text editor. There is also an additional error that says "too much recursion" and it occurs even when the entire attachments/images in rich text are removed.

![image](https://user-images.githubusercontent.com/46025304/229973388-3d8e3fcc-3e48-4026-9833-c3b4f5f6760c.png)

(figure no. 2)

[Screencast from 2023-04-05 11-30-01.webm](https://user-images.githubusercontent.com/46025304/229978432-0299051c-35a5-4f38-9b9d-ed2f2875c5ac.webm)

(video no. 1)

Link to the 21mb image file: https://sample-videos.com/img/Sample-jpg-image-20mb.jpg
other source of test images: https://7132.com/en/corporate/press/high-resolution-images

Bottom line: although we have a 25mb limit, for rich text mode, adding an attachment with a file size of 20 MB and above returns the "attachment is too large" error. adding a 15mb image works but 2 image (15mb+4mb) attachment return "too much recursion" error. Furthermore, when dragging and dropping attachments in rich text mode, I think we shouldn't be placing the image in the rich text editor and adding it as an attachment once at the same time.

**Comment by tomholub** [comment]
Can you confirm that you can successfully send a single attachment of size exactly 20MiB? 20,000,000 bytes.

**Comment by martgil** [comment]
I'm unable to send a single attachment of exactly 20MiB in file size.

![image](https://user-images.githubusercontent.com/46025304/232266822-4b8fe282-2c23-40dd-9d1e-66d36eeaff7f.png)

No issue occurs when sending the same file in non-rich text mode:

![image](https://user-images.githubusercontent.com/46025304/232266936-7205cfdf-c3f8-4570-a534-995c843a18ae.png)


Pdf file I used as attachment: https://research.nhm.org/pdfs/10592/10592.pdf

**Comment by tomholub** [comment]
How about 19 mb, 18 and so on? What is the max?

**Comment by martgil** [comment]
In my testing case, a file that sizes ~19MiB (18.9MiB) and the email went through. I also tried to send an 18MiB file (~18.4) and the email get sent too.

I believe we could consider 18MiB is the safer max option.

18.9MiB file: [file.txt](https://github.com/FlowCrypt/flowcrypt-browser/files/12079094/file.txt)
18.4MiB file: https://research.nhm.org/pdfs/13268/13268.pdf

**Comment by tomholub** [comment]
Ok, please limit to 19mb, only when rich text messages are selected, and link this comment.
